app/controllers/PartTimeHoursController.scala

```diff
@@ -0,0 +1,84 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import cats.data.Validated.{Invalid, Valid}
+import controllers.actions._
+import forms.PartTimeHoursFormProvider
+import javax.inject.Inject
+import models.{PartTimeHours, Period, Periods}
+import navigation.Navigator
+import pages.{PartTimeHoursPage, PartTimePeriodsPage}
+import play.api.i18n.MessagesApi
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import repositories.SessionRepository
+import views.html.PartTimeHoursView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class PartTimeHoursController @Inject()(
+  override val messagesApi: MessagesApi,
+  sessionRepository: SessionRepository,
+  val navigator: Navigator,
+  identify: IdentifierAction,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
+  formProvider: PartTimeHoursFormProvider,
+  val controllerComponents: MessagesControllerComponents,
+  view: PartTimeHoursView
+)(implicit ec: ExecutionContext)
+    extends BaseController {
+
+  val form = formProvider()
+
+  def onPageLoad(idx: Int): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+    getRequiredAnswerOrRedirectV(PartTimePeriodsPage) { partTimePeriods =>
+      withValidPartTimePeriod(partTimePeriods, idx) { partTimePeriod =>
+        val preparedForm = request.userAnswers.getV(PartTimeHoursPage, Some(idx)) match {
+          case Invalid(e)   => form
+          case Valid(value) => form.fill(value.hours)
+        }
+
+        Future.successful(Ok(view(preparedForm, partTimePeriod, idx)))
+      }
+    }
+  }
+
+  def onSubmit(idx: Int): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+    getRequiredAnswerOrRedirectV(PartTimePeriodsPage) { partTimePeriods =>
+      withValidPartTimePeriod(partTimePeriods, idx) { partTimePeriod =>
+        form
+          .bindFromRequest()
+          .fold(
+            formWithErrors => Future.successful(BadRequest(view(formWithErrors, partTimePeriod, idx))),
+            value =>
+              for {
+                updatedAnswers <- Future.fromTry(
+                                   request.userAnswers.set(PartTimeHoursPage, PartTimeHours(partTimePeriod.end, value), Some(idx)))
+                _ <- sessionRepository.set(updatedAnswers)
+              } yield Redirect(navigator.nextPage(PartTimeHoursPage, updatedAnswers, Some(idx)))
+          )
+      }
+    }
+  }
+
+  private def withValidPartTimePeriod(partTimePeriods: Seq[Periods], idx: Int)(f: Period => Future[Result]): Future[Result] =
+    partTimePeriods.lift(idx - 1) match {
+      case Some(period: Periods) => f(period.period)
```

Comment:
I think the view wants to take a type Periods so that we can match on whether it's full or partial and display the accurate to and from dates constrained to the month boundaries